WalletWasabi/Crypto/Randomness/IWasabiRandom.cs

```diff
@@ -8,5 +9,25 @@ namespace WalletWasabi.Crypto.Randomness
 {
 	public interface IWasabiRandom : IRandom
 	{
+		public byte[] GetBytes(int length)
+		{
+			Guard.MinimumAndNotNull(nameof(length), length, 1);
+			var buffer = new byte[length];
+			GetBytes(buffer);
+			return buffer;
+		}
+
+		public Scalar GetScalarNonZero()
```

Comment:
What is this for? Scalars should be {0, N-1}  

It is clear that there are some situations where 0 is not acceptable, for example, when the scalar is used as a private key because  `0 * G` is a point at `Infinity` (Also 1 could be undesirable in this case because `1 * G = G`) however, except in those special cases `0` is a valid number and `0 * G` is a valid point (it is the neutral element so, it is a really important point). In fact WabiSabi uses this in the proof of null.

Reply:
What about this:

```csharp
                Scalar GetRandomScalar(bool allowZero = true)
		{
			Scalar randomScalar;
			int overflow;
			Span<byte> buffer = stackalloc byte[32];

			do
			{
				GetBytes(buffer);
				randomScalar = new Scalar(buffer, out overflow);
			} while (overflow != 0 || (!allowZero && randomScalar.IsZero));

			return randomScalar;
		}
```

It would be more re-usable.

Reply:
We talked with @lontivero about this. I was following @lontivero's code here: https://github.com/lontivero/Wabisabi/blob/master/Wabisabi/Crypto.cs#L210 and that's the only reason I excluded zero, but he said he didn't intend to so I enabled zero.